turbot/__init__.py

```diff
@@ -588,36 +588,32 @@ def fossilsearch_command(self, channel, author, params):
         invalid = items - valid
 
         fossils = load_fossils()
-        # I think we should still show if the searcher needs the fossil being
-        # searched for, just in case they missed it.
-        # theirs = fossils[fossils.author != author.id]
-        theirs = fossils
-        them = theirs.author.unique()
+        users = fossils.author.unique()
         results = defaultdict(list)
         for fossil in valid:
-            havers = theirs[theirs.name == fossil].author.unique()
-            needers = np.setdiff1d(them, havers).tolist()
+            havers = fossils[fossils.name == fossil].author.unique()
+            needers = np.setdiff1d(users, havers).tolist()
             for needer in needers:
                 name = discord_user_from_id(channel, needer)
                 results[name].append(fossil)
 
-        if not results:
-            if len(invalid) > 0:
-                lines = [s("fossil_bad", items=", ".join(sorted(invalid)))]
-            else:
-                lines = [s("fossilsearch_noneed")]
+        if not results and not invalid:
+            return s("fossilsearch_noneed"), None
 
+        if not results and invalid:
+            lines = [s("fossil_bad", items=", ".join(sorted(invalid)))]
+            if valid:
+                lines.append(
+                    s("fossilsearch_row", name="No one", fossils=", ".join(sorted(valid)))
+                )
             return "\n".join(lines), None
 
         lines = [s("fossilsearch_header")]
         for name, needed in results.items():
             need_list = fossils = ", ".join(sorted(needed))
             lines.append(s("fossilsearch_row", name=name, fossils=need_list))
-
-        if len(invalid) > 0:
-            lines.append(" ")
+        if invalid:
```

Comment:
I generally don't like these checks without specifying what you are actually checking even if they are more pythonic.  I find them less clear.  Personally I vastly prefer having the length check visible.  Also prevents unexpected behaviour if invalid isn't the type we think it is for some reason.

You should be glad I'm not doing " 0 != len(invalid)" to ensure there aren't accidental assignments...